src/zenml/pipelines/base_pipeline.py

```diff
@@ -477,6 +477,19 @@ def run(
                 pipeline=self, stack=stack, run_configuration=run_config
             )
 
+            # Override `enable_cache` of all steps if set at run level
+            if enable_cache is not None:
```

Comment:
I think this can be improved:
1) It doesn't respect values for `enable_cache` set from a config file. The combined value from config file and code would be `run_config.enable_cache`
2) The `Compiler` class currently takes care of doing all the merging of pipeline config/step config/run config, I suggest we move this there

Reply:
`Compiler._apply_run_configuration` would probably the best place to do it

Reply:
Makes sense, moved that to `Compiler._apply_run_configuration` now and cleaned it up, feels much better 👍